misk-mcp/src/main/kotlin/misk/mcp/McpTool.kt

```diff
@@ -543,13 +540,6 @@ abstract class StructuredMcpTool<I : Any, O : Any> : McpTool<I>() {
   }
 
   private val outputClass: KClass<O> by lazy {
-    @Suppress("UNCHECKED_CAST")
-    this::class.supertypes
-      .first { type ->
-        (type.classifier as? KClass<*>)?.simpleName?.let { simpleName ->
-          simpleName == StructuredMcpTool::class.simpleName
-        } ?: false
-      }
-      .arguments.last().type!!.classifier as KClass<O>
+    resolveTypeArgument(this::class, StructuredMcpTool::class, 1).classifier as KClass<O>
```

Comment:
very nice, even better if we take the 2nd param (McpTool/StructuredMcpTool) as an inline reified parameter to cut out some of the `::class`

Reply:
Done, and also converted the first param to a receiver, so now it looks like:

```
this::class.resolveTypeArgument<McpTool<*>>(0).classifier as KClass<I>
```